sympy/polys/polyroots.py

```diff
@@ -929,255 +931,256 @@ def roots(f, *gens,
     .. [1] https://en.wikipedia.org/wiki/Cubic_equation#Trigonometric_and_hyperbolic_solutions
 
     """
-    from sympy.polys.polytools import to_rational_coeffs
-    flags = dict(flags)
+    if filter not in (None, 'Z', 'Q', 'R', 'I', 'C'):
+        raise ValueError("Invalid filter: %s" % filter)
+
+    handlers = {
+        'Z': lambda r: r.is_Integer,
+        'Q': lambda r: r.is_Rational,
+        'R': lambda r: all(a.is_real for a in r.as_numer_denom()),
```

Comment:
`fraction(x)` is more literal/naive than `x.as_numer_denom()` - which do you want?